### ISSUE TITLE
Feature request: show Alpine package versions `-rN` in logs

### ISSUE THREAD

**Issue by the-hotmann** [qdm12/gluetun#2350]
### What's the feature 🧐

Regular Updates when dependencies, or base gets updated.

Currently (2024.07.27) this is the setup:

|Name|Current Version|Version available|Where to be found|
|---|---|---|---|
|**`Alpine`**|`3.19.1`|`3.19.3` (Update)|[`LINK`](https://alpinelinux.org/releases/)|
|**`Alpine`**|`3.19.1`|`3.20.2` (Upgrade)|[`LINK`](https://alpinelinux.org/releases/)|
|**`OpenVPN v2.5`**|`2.5.8`|`2.5.10-r1`|[`LINK`](https://pkgs.alpinelinux.org/packages?name=openvpn*&branch=v3.17&repo=&arch=&maintainer=)|
|**`OpenVPN v2.6`**|`2.6.8`|`2.6.11-r0`|[`LINK`](https://pkgs.alpinelinux.org/packages?name=openvpn&branch=v3.20&repo=&arch=&maintainer=)|
|**`Unbound`**|`1.20.0`|`1.20.0-r0`|[`LINK`](https://pkgs.alpinelinux.org/packages?name=unbound&branch=v3.20&repo=&arch=&maintainer=)|
|**`IPtables`**|`1.8.10`|`1.8.10-r3`|[`LINK`](https://pkgs.alpinelinux.org/packages?name=iptables&branch=v3.20&repo=&arch=&maintainer=)|

Please also show the `-r*` addition, as this would make the versions more transparent, if there are security issues with specific version we can contact you and initiate a rebuild with the fixed version.

Thanks for your nice tool! :)

**Comment by the-hotmann** [comment]
@qdm12 thanks very much for updating the base+dependencies.

Is there a possibility, you also show the [Subsequent package versions](https://wiki.alpinelinux.org/wiki/Package_policies#:~:text=Subsequent%20package%20versions%20are%20%2Dr0%2C%20%2Dr1%2C%20and%20so%20on%3B%20the%20number%20is%20from%20%24pkgver.) of each dependency for transparency reasons?

Thank you very much!

**Comment by qdm12** [comment]
In the end I did upgrade to Alpine 3.20, because that iptables issue is affecting some machines not supporting iptables-legacy (see https://github.com/qdm12/gluetun/issues/2304#issuecomment-2254540211). So now everything should be up to date. There is still  a discussion to have about showing Alpine package versions `-rN` though 🤔

**Comment by the-hotmann** [comment]
> since upgrading these often break the setups for a good chunk of users.

Yes, indeed, many Synology users (myself included) are experiencing problems with the new 3.20.x upgrade. However, I hope this can be fixed. Otherwise, older Synology devices will need to remain on 3.19.x.

> Well this is specific to Alpine, and the long-term goal is to NOT depend on an OS

That would be fantastic! I can imagine this application being solely based on Golang and compiled with `CGO_ENABLED=0` into a single static binary, which would not require any other OS. It could potentially be shipped in a Docker-Scratch Docker image.

> I'm unaware to be honest, what's the difference in versioning between v1.8.1-r0 and v1.8.1-r1? Isn't the bug fix the last `1` in `v1.8.1`?

Yes and no. Once `1.8.1` is released by the software maintainer, a package maintainer will compile it from scratch (source build) and release it as the Alpine package `1.8.1-r0` (first release). If a security issue is found in `1.8.1` and a fix is released on GitHub, there isn't always an immediate new release. However, the package maintainer can rebuild it and release it as `1.8.1-r1`, allowing everyone on Alpine to apply the fix immediately.
Alpine's package manager also allows insecure or outdated releases to be flagged, notifying the package maintainer who will then release a fixed version (if a fix is available). You can identify these flagged releases by checking the release color (green = up-to-date, red = flagged). [Red Example](https://pkgs.alpinelinux.org/package/edge/main/x86_64/openssl)

In short: the official version (semver) is defined by the software maintainer, while the `-rN` version is defined by the package maintainer. As long as it is based on Alpine, I appreciate the transparency of the `-rN` info.

**Comment by the-hotmann** [comment]
About the Synology Problem: I guess on the long run this will not be solvable, since Synology does not provide Kernel Upgrades.
And since Golang (as many other vendors, OSs etc..) do cut support for old kernels this anyway will be a problem on the long run. But I guess for now just using `iptables-legacy` as fallback will be enough as it worked on `3.19.3`?

Anyway on all my other machines (Debian SID | Kernel v`6.9.12-1`) the new version works flawlessly. But keep in mind, that I have not checked if it really is secure. I assume you make such tests for all releases as some type of A/B-Testing?


Thanks again for your hard work and the chat :)

**Comment by qdm12** [comment]
> Yes, indeed, many Synology users (myself included) are experiencing problems with the new 3.20.x upgrade.

What problems are you experiencing?? Feel free to comment here or create other issues per problem.

> Alpine's package manager also allows insecure or outdated releases to be flagged, notifying the package maintainer who will then release a fixed version (if a fix is available). You can identify these flagged releases by checking the release color (green = up-to-date, red = flagged).

I did... not know that!! Thanks! 👍 

> [#ref](https://github.com/golang/go/issues/67001)
> So my old Syno (DS1817+) anyway soon will very likely not run anything that is based on Golang >`1.24.x`. But that is my problem.

Did not know that either, interesting! 🎖️ As in, I can maintain 2 versions in that case, there would be a oldkernel tag or something, especially since Go is relatively compatible even from a higher version to a lower one. Anyway, we'll see later!

> the official version (semver) is defined by the software maintainer, while the -rN version is defined by the package maintainer. As long as it is based on Alpine, I appreciate the transparency of the -rN info.

Ok I guess then this information could be useful. Even better would be a check against Alpine database to check if it's 'green' or 'red', but that's when the tunnel is up later. Could be nice to report 'red' packages 👍 I'll look into that instead, so just printing the versions as it does now, but after the tunnel is up, report any "red" packages. Thoughts on that?

> But keep in mind, that I have not checked if it really is secure. I assume you make such tests for all releases as some type of A/B-Testing?

Define "secure" 😄 
As in Gluetun is a big-big wrapper around OpenVPN or Wireguard; security depends on those protocols and especially on the VPN servers configuration (especially for OpenVPN). Some providers are quite bad at having a decent configuration, so that's that. Well there is the firewall in Gluetun (and soon, I gotta do #1697), so these are security risks indeed, but they don't change much (I hate fiddling with those now that it works 😄).
I don't really do any testing to compare releases (ain't no time for that); but I do have my little release process, more or less described at the bottom of https://github.com/qdm12/gluetun-wiki/blob/main/setup/docker-image-tags.md#why-is-latest-pointing-to-the-edge-of-development if that is relevant.

**Comment by the-hotmann** [comment]
> What problems are you experiencing?? Feel free to comment here or create other issues per problem.

Sorry should have updated the post. https://github.com/qdm12/gluetun/issues/2375#issuecomment-2255080378 fixed the issue.

> Did not know that either, interesting! 🎖️ As in, I can maintain 2 versions in that case, there would be a oldkernel tag or something, especially since Go is relatively compatible even from a higher version to a lower one. Anyway, we'll see later!

That probably would be neat in the future. For me: I will switch the container to the Intel NUC and have it run on the newest kernel, but since I read some other issues, I noticed a lot of people using Synologys - if they use old ones (just like me) that would adress that issue :)

> Thoughts on that?

Will test it and report back :)
Also: flagging AFAIK is just allowed/possible in the edge-branch, since all other branches, are expected to be "outdated". So maybe this does not work as initially intended..

> Define "secure" 😄

Of course this container is based on other applications/protocols. If they are unsecure the container will be, but the responibility/root would not be found here.
I was more thinking of network, ip/dns-leaks and routs.

In general my thoughts have been:
1. is the container shipped with ALL routs removed, so that even in a critical race-condition another container (which may starts/boots faster) can not leak its IP?
Or are the routs only removed by executing the entrypoint/entry-script and therefore opening a possible window for such race-condition?
2. is the container malfunctions (just like some days ago) is there a way of it booting up and letting traffic out with the real IP?
3. in cases of OOM/crash etc. are there any potential leaks?

A lot of questions, but I genrally would like to understand how the "start-up" and "shut-down" proceedure works, and how it would behave in a crash/faulty state :)

Thanks again!

**Comment by qdm12** [comment]
I migrated your question to #2370 and see my answer on https://github.com/qdm12/gluetun/discussions/2370#discussioncomment-10216090 😉
